diff --git a/vllm/models/deepseek_v4/nvidia/flashmla.py b/vllm/models/deepseek_v4/nvidia/flashmla.py
index 29ec2d6f2abf..f0b00163d1a6 100644
--- a/vllm/models/deepseek_v4/nvidia/flashmla.py
+++ b/vllm/models/deepseek_v4/nvidia/flashmla.py
@@ -96,8 +96,11 @@ def forward_mqa(
                 // self.compress_ratio
             )
             M = N + self.window_size + self.max_num_batched_tokens
-            assert self.topk_indices_buffer is not None
-            top_k = 0 if swa_only else self.topk_indices_buffer.shape[-1]
+            if swa_only:
+                top_k = 0
+            else:
+                assert self.topk_indices_buffer is not None
+                top_k = self.topk_indices_buffer.shape[-1]
             combined_topk = round_up(top_k + self.window_size, 128)
             current_workspace_manager().get_simultaneous(
                 ((self.PREFILL_CHUNK_SIZE, M, q.shape[-1]), torch.bfloat16),
